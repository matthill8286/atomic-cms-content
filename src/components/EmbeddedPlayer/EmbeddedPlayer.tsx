import React from 'react'
import { EmbeddedPlayerProps } from './EmbeddedPlayer.interface'

export const EmbeddedPlayer: React.FC<EmbeddedPlayerProps> = ({
  title = '',
  width = '100%',
  height = '100%',
  embedUrl,
  frameBorder = '0',
  features = ['picture-in-picture', 'showinfo=0'],
  allowfullscreen,
  ...props
}) => {
  return (
    <iframe
      title={title}
      width={width}
      height={height}
      src={embedUrl}
      frameBorder={frameBorder}
      allow={features.join()}
      allowFullScreen={allowfullscreen}
      {...props}
    />
  )
}
