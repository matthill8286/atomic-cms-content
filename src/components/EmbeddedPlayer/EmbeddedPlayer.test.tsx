import React from 'react'
import { renderWithTheme } from '@matthill8286/atomic-ui'
import { EmbeddedPlayer } from './EmbeddedPlayer'

describe('EmbeddedPlayer', () => {
  it('shows a youtube video', () => {
    const tree = renderWithTheme(
      <EmbeddedPlayer embedUrl="https://www.youtube.com/embed/TeQ_TTyLGMs" />
    )
    expect(tree).toMatchSnapshot()
  })
})
