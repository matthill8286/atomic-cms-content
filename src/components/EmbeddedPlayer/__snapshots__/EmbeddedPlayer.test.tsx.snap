// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedPlayer shows a youtube video 1`] = `
<iframe
  allow="picture-in-picture,showinfo=0"
  frameBorder="0"
  height="100%"
  src="https://www.youtube.com/embed/TeQ_TTyLGMs"
  title=""
  width="100%"
/>
`;
